### PR TITLE
[5.0] Fail with an error if key:generate fails.

### PR DESCRIPTION
**Pull request by jaden**
First, attempt to copy .env.example to avoid failing.

Fix for issue #7356.

### CODE REVIEW

**Comment by Arrilot** [comment]
In my opinion `php artisan key:generate` should not silently create a .env file instead of a developer

Simple error and exit like that is fine imho
```php 
if ( ! file_exists(base_path('.env')))
{
        $this->error("<Some error about missing .env file here");
}
```

**Comment by GrahamCampbell** [comment]
I agree with @Arrilot.

**Comment by jaden** [comment]
I thought it was just extra helpful, but I can see your side too. I pushed just the error and exit.

**Comment by GrahamCampbell** [comment]
Just push to your existing branch. Your changes will appear here. :)

**Comment by jaden** [comment]
I already pushed the second change. Is there something else I need to do?

**Comment by jaden** [comment]
This is marked as required changes, but I don't know what other changes are required. I've already made the requested modification to the original commit.

File: src/Illuminate/Foundation/Console/KeyGenerateCommand.php [hunk 1]
```diff
@@ -36,13 +36,16 @@ public function fire()
...
 		$path = base_path('.env');
 
-		if (file_exists($path))
+		if ( ! file_exists($path))
 		{
-			file_put_contents($path, str_replace(
-				$this->laravel['config']['app.key'], $key, file_get_contents($path)
-			));
+			$this->error("A .env file needs to be created first.");
+			exit(1);
```
Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/KeyGenerateCommand.php]: No. You 100% can't do that!

Comment (jaden) [inline src/Illuminate/Foundation/Console/KeyGenerateCommand.php]: Thanks for the update. Are you saying not to call exit?

Comment (GrahamCampbell) [inline src/Illuminate/Foundation/Console/KeyGenerateCommand.php]: Yes. We can't do that. Just `return 1`. Symfony deals with evrything else.

**Comment by jaden** [comment]
D'oh. I just pushed the fix to my branch but @taylorotwell closed it.

**Comment by sbusch** [comment]
@taylorotwell Any chance for this to be reopened / merged? I just stumbled over the same problem and an appropriate error message would have helped.

**Pull request closed without merging**